WARNING: No route found for IPv6 destination :: (no default route?)
/home/scada/workspace/scada-testbed/automatized_tests/attacker/mitm_gui.py:14: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import GLib, Gtk, GObject
Starting attacker, interface: eno1
#####################################################################################################
INITIALIZING
Attacking CONTROLLER ip: 192.168.2.2 mac: b8:27:eb:c1:cb:21 and CAR ip: 192.168.2.3 mac: 6c:19:8f:01:a9:1d
IDLE
POISONING
[Request] Enable system forwarding
[LOG_LEARNING] WAITING TO REACH LIMIT
[LOG_LEARNING] WAITING TO REACH LIMIT
[LOG_LEARNING******] LIMIT REACHED
[LOG_LEARNING] START SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCING
[LOG_LEARNING] SYNCED START RECORDING
[LOG_LEARNING] Direction Change! New threshold 59
[LOG_LEARNING] Direction Change! New threshold 64
========================= **ATTACKER INITIALIZIZING ATTACK AT: 13:59:48
[ATTACKING] INTERCEPTED TO: 192.168.2.2 FUNC: FUNC_READ_COIL FORWARDING TO: 192.168.2.3 FUNC: FUNC_READ_SR
Packet from: 192.168.2.3 command: FUNC_READ_COIL DROPPED
[ATTACKING] INTERCEPTED TO: 192.168.2.3 FUNC: FUNC_READ_SR FORWARDING TO: 192.168.2.2 FUNC: FUNC_READ_SR
[ATTACKING] INTERCEPTED TO: 192.168.2.3 FUNC: FUNC_READ_COIL FORWARDING TO: 192.168.2.3 FUNC: FUNC_READ_COIL
[ATTACKING] INTERCEPTED TO: 192.168.2.3 FUNC: FUNC_READ_SR FORWARDING TO: 192.168.2.2 FUNC: FUNC_READ_COIL
Traceback (most recent call last):
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/mitm_gui.py", line 553, in <module>
    sniff(filter="tcp and port " + str(ModbusPort) + " and not ether src " + cs.attackerMAC, stop_filter=handlePacketAttacking, store=0, count=MaxCount)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/sendrecv.py", line 620, in sniff
    if stop_filter and stop_filter(p):
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/mitm_gui.py", line 391, in handlePacketAttacking
    pkt_funcCode = pkt.funcCode
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 191, in __getattr__
    fld,v = self.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 188, in getfield_and_val
    return self.payload.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 188, in getfield_and_val
    return self.payload.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 188, in getfield_and_val
    return self.payload.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 188, in getfield_and_val
    return self.payload.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 188, in getfield_and_val
    return self.payload.getfield_and_val(attr)
  File "/home/scada/workspace/scada-testbed/automatized_tests/attacker/scapy/packet.py", line 1104, in getfield_and_val
    raise AttributeError(attr)
AttributeError: funcCode
AUTOTEST FAILURE
